### PR TITLE
Add async when announcements are send by message

### PR DESCRIPTION
**Pull request by Zowac**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Fixed issues | #1580

In the continuity to put some tasks in asynchronous: 
added asynchronous when an announcement is sent via message

### CODE REVIEW

File: src/main/core/Messenger/Message/SendMessage.php [hunk 1]
```diff
@@ -19,12 +19,14 @@ class SendMessage
     private $content;
     private $object;
     private $users;
+    private $sender;
 
-    public function __construct(string $content, string $object, array $users)
+    public function __construct(string $content, string $object, array $users, User $sender = null)
```
Comment (chalasr) [inline src/main/core/Messenger/Message/SendMessage.php:24]:
```suggestion
    public function __construct(string $content, string $object, array $users, ?User $sender = null)
```

File: src/plugin/announcement/Messenger/Message/SendAnnouncement.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    private $receivers;
+    private $announcementId;
+    private $sender;
+
+    public function __construct(
+        string $content,
+        string $object,
+        array $receivers,
+        int $announcementId,
+        User $sender = null
```
Comment (chalasr) [inline src/plugin/announcement/Messenger/Message/SendAnnouncement.php:20]:
```suggestion
        ?User $sender = null
```

File: src/plugin/announcement/Manager/AnnouncementManager.php [hunk 3]
```diff
@@ -68,21 +72,13 @@ public function sendMessage(Announcement $announcement, array $users = [])
         $this->om->persist($announcementSend);
         $this->om->flush();
```
Comment (chalasr) [inline src/plugin/announcement/Manager/AnnouncementManager.php:73]: @Elorfin could we move this logic to the async handler?

Comment (Elorfin) [reply src/plugin/announcement/Manager/AnnouncementManager.php:73]: Yes we can. This is just a log for sent messages. NB. this should be replaced later with generic mail logs.

File: src/plugin/announcement/Repository/AnnouncementSendRepository.php [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace Claroline\AnnouncementBundle\Repository;
+
+use Doctrine\ORM\EntityRepository;
+
+class AnnouncementSendRepository extends EntityRepository
+{
+}
```
Comment (Elorfin) [inline src/plugin/announcement/Repository/AnnouncementSendRepository.php:9]: You don't need this if you don't declare custom methods.

File: src/plugin/announcement/Manager/AnnouncementManager.php (hunk 3 above)
Comment (Zowac) [reply src/plugin/announcement/Manager/AnnouncementManager.php]: Done, thx :)

File: src/plugin/announcement/Messenger/SendAnnouncementHandler.php [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            [
+                $sendAnnouncement->getContent(),
+                $sendAnnouncement->getObject(),
+                $sendAnnouncement->getReceivers(),
+                $sendAnnouncement->getSender(),
+            ]
+        );
+
+        //it's kind of a hack because this is not using the crud... but wathever.
+        $this->eventDispatcher->dispatch('crud.post.create.announcement_send', 'Claroline\\AppBundle\\Event\\Crud\\CreateEvent', [
```
Comment (chalasr) [inline src/plugin/announcement/Messenger/SendAnnouncementHandler.php:58]: `CreatedEvent::class` (+ `use` statement)

**Review by chalasr — APPROVED** [review]
Looks good 👍

**Review by Elorfin — APPROVED** [review]
👍

**Pull request merged**